pkg/katapult/virtual_machine_group_test.go

```diff
@@ -0,0 +1,169 @@
+package katapult
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestVirtualMachineGroup_JSONMarshaling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *VirtualMachineGroup
+	}{
+		{
+			name: "empty",
+			obj:  &VirtualMachineGroup{},
+		},
+		{
+			name: "full",
+			obj: &VirtualMachineGroup{
+				ID:        "id",
+				Name:      "name",
+				Segregate: true,
+				CreatedAt: timestampPtr(934834834),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func TestVirtualMachineGroupsClient_List(t *testing.T) {
+	// Correlates to fixtures/virtual_machine_groups_list*.json
+	virtualMachineGroupList := []*VirtualMachineGroup{
+		{
+			ID:        "vmgrp_gsEUFPp3ybVQm5QQ",
+			Name:      "vm group 1",
+			Segregate: true,
+		},
+		{
+			ID:        "vmgrp_bcfdEFn2viWCm5ve",
+			Name:      "vm group 2",
+			Segregate: false,
+		},
+	}
+
+	type args struct {
+		ctx context.Context
+		org *Organization
+	}
+	tests := []struct {
+		name       string
+		args       args
+		want       []*VirtualMachineGroup
+		wantQuery  *url.Values
+		errStr     string
+		errResp    *ResponseError
+		respStatus int
+		respBody   []byte
+	}{
+		{
+			name: "by organization ID",
+			args: args{
+				ctx: context.Background(),
+				org: &Organization{ID: "org_O648YDMEYeLmqdmn"},
+			},
+			want: virtualMachineGroupList,
+			wantQuery: &url.Values{
+				"organization[id]": []string{"org_O648YDMEYeLmqdmn"},
+			},
+			respStatus: http.StatusOK,
+			respBody:   fixture("virtual_machine_groups_list"),
+		},
+		{
+			name: "invalid API token response",
+			args: args{
+				ctx: context.Background(),
+				org: &Organization{ID: "org_O648YDMEYeLmqdmn"},
+			},
+			errStr:     fixtureInvalidAPITokenErr,
+			errResp:    fixtureInvalidAPITokenResponseError,
+			respStatus: http.StatusForbidden,
+			respBody:   fixture("invalid_api_token_error"),
+		},
+		{
+			name: "non-existent organization",
+			args: args{
+				ctx: context.Background(),
+				org: &Organization{ID: "org_O648YDMEYeLmqdmn"},
+			},
+			errStr:     fixtureOrganizationNotFoundErr,
+			errResp:    fixtureOrganizationNotFoundResponseError,
+			respStatus: http.StatusNotFound,
+			respBody:   fixture("organization_not_found_error"),
+		},
```

Comment:
Test cases for `organization_not_activated`, `organization_suspended`, and `permission_denied` as detailed in the  [API docs](https://developers.katapult.io/api/docs/latest/routes/organizations/:organization/virtual_machine_groups/get/) are needed here too.

Generally we should try to have test cases for all possible types of errors that the API can return. This will become even more important once #64 is resolved and we have concrete structs in Go for each error type.

It's very possible some error types are missing from some existing tests though, that's simply an oversight on my part, or it's a newly added error type like [`organization_not_activated`](https://developers.katapult.io/api/docs/latest/errors/cd63baeb6b/) which was added to Katapult in early February.